weldx/core.py

```diff
@@ -0,0 +1,232 @@
+                raise Exception(
+                    "The mathematical expression must have exactly 1 free "
+                    "variable that represents time."
+                )
+            time_var_name = data.get_variable_names()[0]
+            try:
+                eval_data = data.evaluate(**{time_var_name: Q_(1, "second")})
+                self._units = eval_data.units
+                if isinstance(eval_data.magnitude, np.ndarray):
+                    self._shape = eval_data.magnitude.shape
+                else:
+                    self._shape = tuple([1])
+            except pint.errors.DimensionalityError:
+                raise Exception(
+                    'Expression can not be evaluated with "pint.Quantity(1, "seconds")"'
+                    ". Ensure that every parameter posses the correct unit."
+                )
+
+            self._data = data
+            self._time_var_name = time_var_name
+
+            try:
+                self.interp_time(Q_([1, 2], "second"))
+                self.interp_time(Q_([1, 2, 3], "second"))
+            except Exception as e:
+                raise Exception(
+                    "The expression can not be evaluated with arrays of time deltas. "
+                    "Ensure that all parameters that are multiplied with the time "
+                    "variable have an outer dimension of size 1. This dimension is "
+                    "broadcasted during multiplication. The original error message was:"
+                    f' "{str(e)}"'
+                )
+
+        else:
+            raise TypeError(f'The data type "{type(data)}" is not supported.')
+
+    @property
+    def data(self):
+        if isinstance(self._data, xr.DataArray):
+            return self._data.data
+        return self._data
+
+    @property
+    def interpolation(self):
+        return self._interpolation
+
+    @property
+    def time(self):
+        if isinstance(self._data, xr.DataArray):
+            if len(self._data.time) == 1:
+                return None
+            return ut.to_pandas_time_index(self._data.time.data)
+        return self._time
+
+    def interp_time(self, time):
+        if isinstance(self._data, xr.DataArray):
+            if self._interpolation == "linear":
+                return ut.xr_interp_like(
+                    self._data,
+                    {"time": time},
+                    assume_sorted=False,
+                    broadcast_missing=False,
+                )
+            raise Exception("not implemented")
+
+        if not isinstance(time, Q_) or not time.check(UREG.get_dimensionality("s")):
+            raise ValueError('"time" must be a time quantity.')
+
+        if len(self.shape) > 1 and isinstance(time.magnitude, np.ndarray):
+            while len(time.magnitude.shape) < len(self.shape):
+                time = Q_(time.magnitude[:, np.newaxis], time.units)
+
+        time = {self._time_var_name: time}
+        return self._data.evaluate(**time)
```

Comment:
this should work if we want to return `xarray.DataArray` for all cases:
```python
        data = self._data.evaluate(**{self._time_var_name: time})
        if not np.iterable(data.magnitude):  # make sure quantity is not scalar value
            data = data * np.array([1])
        if hasattr(time, "shape"):  # squeeze out any helper dimensions
            time = np.squeeze(time)
        time = ut.to_pandas_time_index(time)
        dax = xr.DataArray(data=data) # don't know exact dimensions so far
        dax = dax.rename({"dim_0": "time"}).assign_coords({"time": time})
        return dax
```

Needs updated `to_pandas_time_index` for `pint.Quantity`

Reply:
Done